app/workers/UploadQueue/__init__.py

```diff
@@ -418,7 +424,11 @@ def _handle_failure(msg: func.QueueMessage, scan_report_id: str) -> None:
     logger.info(f"dequeue_count {msg.dequeue_count}")
 
     if msg.dequeue_count == 2:
-        update_scan_report_status(scan_report_id, UploadStatusType.FAILED)
+        update_job(
+            JobStageType.UPLOAD_SCAN_REPORT,
+            StageStatusType.FAILED,
+            scan_report=ScanReport.objects.get(id=scan_report_id),
```

Comment:
(this is in a few places)

Reply:
I tried to use that but it didn't go through. The reason, I think, is the function `update_job` was designed to take scanReport instance, not SR id, then use this to filter and get the desired Job record, which has scan_report field needing to be SR instance as well (because we wanted the `on_delete=models.CASADE` function)

Reply:
nice - thanks for explaining